app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx

```diff
@@ -243,13 +294,13 @@ const TransactionDetails: React.FC<Props> = ({ order }: Props) => {
 					</ListItem.Body>
 					<ListItem.Amount>
 						<Text small bold primary>
-							{order.data.fiatCurrency.denomSymbol}
-							{order.amount} {order.data.fiatCurrency.symbol}
+							{currencySymbol}
+							{renderFiat(amount, currency, data.fiatCurrency?.decimals)}
 						</Text>
 					</ListItem.Amount>
 				</ListItem.Content>
 				{order.state === SDK_ORDER_STATUS.Completed && (
-					<TouchableOpacity>
+					<TouchableOpacity onPress={() => handleLinkPress(explorer.tx(txHash))}>
```

Comment:
I think `txHash` might be undefined for some providers, can you conditionally render when `txHash` is truthy?